src/actors/MeleeMob.gd

```diff
@@ -66,7 +69,12 @@ func _physics_process(delta) -> void:
 		return
 
 	set_velocity()
-	move_and_collide(velocity * delta)
+	var collision = move_and_collide(velocity * delta)
+	if collision && (is_colliding_with_player_or_enemy(collision.collider)):
+		collision.collider.move_and_slide(velocity * Utils.calculate_resistance(self, collision.collider))
+
+func is_colliding_with_player_or_enemy(collider) -> bool:
+	return collider == player || enemy_node.get_children().has(collider)
```

Comment:
We're looping through all the enemies in the `EnemyNode` every collision here to see if the collider matches any of those?

Reply:
Yes pretty much. If this was a hash lookup it would be a lot better.

Reply:
cleaned this up by adding enemies and players to a group "movable" then just checking if the collision.collider is part of that group